doc/blog/Wasm-JS series/Blog - WASM.md

```diff
@@ -0,0 +1,658 @@
+   this.RegisterHtmlCustomEventHandler("OpenedStateChanged", OnOpenedStateChanged, isDetailJson: false);
+   ```
+
+2. Add the implementation for the two handlers in the class:
+
+   ``` csharp
+   private void OnDateChanged(object sender, HtmlCustomEventArgs e)
+   {
+       if(DateTimeOffset.TryParse(e.Detail, DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AssumeLocal, out var dto))
+       {
+           SelectedDateTime = dto;
+       }
+   }
+   
+   private void OnOpenedStateChanged(object sender, HtmlCustomEventArgs e)
+   {
+       switch(e.Detail)
+       {
+           case "open":
+               IsPickerOpened = true;
+               break;
+           case "closed":
+               IsPickerOpened = false;
+               break;
+       }
+   }
+   ```
+
+3. Change the initialization of `Flatpickr` in injected JavaScript to raise events. Change the implementation of the `OnLoaded` method for this instead:
+
+   ``` csharp
+   private void OnLoaded(object sender, RoutedEventArgs e)
+   {
+       // For demo purposes, Flatpickr is loaded directly from CDN.
+       // Uno uses AMD module loading, so you must give a callback when the resource is loaded.
+       var javascript = $@"require([""https://cdn.jsdelivr.net/npm/flatpickr""], f => {{
+               // Get HTML <input> element
+               const element = document.getElementById(""{HtmlId}"");
+   
+               // Route Flatpickr events following Uno's documentation
+               // https://platform.uno/docs/articles/wasm-custom-events.html
+               const options = {{
+                       onChange: (dates, str) => element.dispatchEvent(new CustomEvent(""DateChanged"", {{detail: str}})),
+                       onOpen: () => element.dispatchEvent(new CustomEvent(""OpenedStateChanged"", {{detail: ""open""}})),
+                       onClose: () => element.dispatchEvent(new CustomEvent(""OpenedStateChanged"", {{detail: ""closed""}}))
+                   }};
+   
+               // Instanciate Flatpickr on the element
+               f(element, options);
+           }});";
+   
+       WebAssemblyRuntime.InvokeJS(javascript);
+   }
+   ```
+
+4. Compile & Run. Here's the result:
+   ![](flatpickr-final.gif)
+
+## 🔬 Going further
+
+This article illustrates how to integrate external assets (javascript and css files) and how to leverage JavaScript's `CustomEvent` in an Uno application.
+
+More steps could be done to make the code more production ready:
+
+* **Make the control multi-platform**. Many DateTime pickers exists on all platforms. It should be easy on other platforms to connect the same control to another greate Date picker native to the platform - no need to embed a WebView for this on other platforms.
+* **Create script files instead of generating dynamic javascript**. As in previous article, this would have the advantage of improving performance and increase the ability to debug it.
```

Comment:
```suggestion
* **Create script files instead of generating dynamic Javascript**. As in previous article, this would have the advantage of improving performance and also improve the debuggability of the code.
```